credentials-manager.js

```diff
@@ -153,8 +157,10 @@ class CredentialsManager {
     credentials[profile].aws_session_expiration = sessionExpiration.toISOString();
     credentials[profile].aws_session_token = sessionToken;
 
-    await fs.mkdir(dirname(path), { recursive: true });
-    await fs.writeFile(path, ini.encode(credentials))
+    if (!fs.existsSync(dirname(path))) {
+      fs.mkdirSync(dirname(path));
```

Comment:
This one is trickier. You've introduced a regression here because mkdir on node 8 does not support the recursive attribute.

Reply:
added function to create the directories recursively